src/error.rs

```diff
@@ -293,6 +293,8 @@ pub(crate) enum Kind {
     UrlBadScheme,
     #[cfg(feature = "default-tls")]
     Tls(::native_tls::Error),
+    #[cfg(feature = "rustls-tls")]
+    Tls(::rustls::TLSError),
```

Comment:
In order to support when both `default-tls` **and** `rustls-tls` are enabled, these should probably become two variants, `NativeTls` and `Rustls`.

Reply:
I don't know if it makes sense to choose TLS backend at runtime, but I'll give it a try.

Reply:
Yea, it may not be the best to support both at runtime, and perhaps it should be changed to only 1 or the other, but I believe it'd be a breaking change. native-tls provides a couple features that rustls never will (disabling security), so we cannot add a non-default feature that doesn't have the same methods.